### PR TITLE
Sourcery poc

### PR DESCRIPTION
**Pull request by joesus**
## Checklist

- [x] I've read the [Contributing Guidelines](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I've completed the [Contributor License Agreement](https://developers.facebook.com/opensource/cla)
- [x] I've ensured that all existing tests pass and added tests (when/where necessary)
- [x] I've ensured that my code lints properly: (`swiftlint && swiftlint autocorrect --format`)
- [x] I've updated the documentation (when/where necessary) and [Changelog](CHANGELOG.md) (when/where necessary)
- [x] I've added the proper label to this pull request (e.g. `bug` for bug fixes)

## Pull Request Details

Initial steps at using Sourcery and Stencil to generate code that will play nice with ObjC. 
### What this is:
* instantiating wrapper classes for structs and enums
* instantiating wrapper classes for classes since they may contain properties that do not work with ObjC
* setting up pass-through properties
* a few dummy classes for experimenting with the stencil templates

### What this is not:
* fully implemented

### What is missing:
* methods
* complex types such as generics
* unit testing wrapper classes

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+./Sourcery/Generated/* linguis-generated
```
Comment (codytwinton) [inline .gitattributes]: https://github.com/github/linguist#generated-code

```suggestion
Sourcery/Generated/* linguist-generated=true
```

File: .sourcery.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+sources:
+  - ./Sources/Core
+templates:
+  - ./Sourcery/Templates
```
Comment (codytwinton) [inline .sourcery.yml]:
```suggestion
  - ./Codegen/Templates
```

File: .sourcery.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+output:
+  ./Sourcery/Generated
```
Comment (codytwinton) [inline .sourcery.yml]:
```suggestion
  ./Codegen/Generated
```

**Comment by joesus** [comment]
@codytwinton - Let me know when this looks ready to merge and I'll do a sweep and remove the dummy project and files. Left them in there to make it easier to review but they should not be included. 👍

**Pull request closed without merging**